### PR TITLE
Social Previews: Decode entities in post titles, excerpts, and descriptions

### LINKED ISSUE #26760: Jetpack Social Preview does not render & properly
### Quick summary

The Jetpack Social Preview does not render & symbol properly in Simple sites. It works fine on Atomic Sites. 

### Steps to reproduce

1. Open a simple site and create a new post or page.
2. Add & symbol in the title.
3. Check Jetpack Social Preview

image

image

### What you expected to happen

Social Preview should render & symbol correctly.

### What actually happened

Social Preview rendered & symbol as `&amp;`

### Context

User report: 24756386-hc

### Platform (Simple, Atomic, or both?)

Simple

### Theme-specific issue?

No

### Browser, operating system and other notes

_No response_

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

No but the platform is still usable

### Workaround details

_No response_

### PR DESCRIPTION
**Pull request by sixhours**
Fixes https://github.com/Automattic/jetpack/issues/26760

## Proposed changes:

* Run `decodeEntities` on post title & description before showing them in the Social Previews panel.
* Core seems to encode special characters in post meta while Jetpack does not.
* ~~I'm posing this as a proof of concept because I'm concerned it's not as secure as it could be. I see we're using `decodeEntities` elsewhere in the plugin, so it's probably okay? But I wonder about the potential for XSS issues sneaking through.~~ Answered in comments.

**Before**

Screenshot 2024-11-19 at 12 18 32 PM

**After**

Screenshot 2024-11-19 at 12 17 06 PM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply this patch to your sandbox and sandbox the API
* Create a new post in the editor on a Simple site with a Business plan. Ensure the post title and description include a special character like `&`
* Publish the post
* Go to the Jetpack toolbar
* Go to Social Previews
* Click Preview to see the modal
* The special characters should be rendered/decoded rather than showing as encoded (like `&amp;`)

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/social-previews/modal.js [hunk 1]
```diff
@@ -37,8 +41,8 @@ const SocialPreviewsModal = function SocialPreviewsModal( { onClose, initialTabN
 						<tab.preview
 							// pass only the props that are common to all previews
 							excerpt={ excerpt }
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/components/social-previews/modal.js]: Can we please do the same for `excerpt`?

**Comment by manzoorwanijk** [comment]
> I'm posing this as a proof of concept because I'm concerned it's not as secure as it could be. I see we're using `decodeEntities` elsewhere in the plugin, so it's probably okay? But I wonder about the potential for XSS issues sneaking through.

Thank you for considering that. Since the text is passed to and rendered by React components, it's safe to decode the HTML entities.

File: projects/js-packages/publicize-components/src/components/social-previews/modal.js (hunk 1 above)
Comment (sixhours) [reply projects/js-packages/publicize-components/src/components/social-previews/modal.js]: Sure thing! All done :)

**Comment by sixhours** [comment]
FYI I'm on vacation next week. If this gets approved before I get back, please feel free to merge on my behalf!

**Review by manzoorwanijk — APPROVED** [review]
This looks good. Thank you for fixing it.

**Pull request merged**